pkg/apis/nfd/v1alpha1/rule.go

```diff
@@ -50,11 +50,6 @@ func (r *Rule) Execute(features feature.Features) (RuleOutput, error) {
 				matched = true
 				utils.KlogDump(4, "matches for matchAny "+r.Name, "  ", matches)
 
-				if r.labelsTemplate == nil {
-					// No templating so we stop here (further matches would just
-					// produce the same labels)
-					break
-				}
```

Comment:
It is still useful to check if the templates have been defined. If there would be 10 items under `MatchAny` and the first matches (and no templates are specified) there is no use to evaluate the 9 remaining items.

My suggestion to change/fix this instead

```
                                if r.LabelsTemplate == "" && r.VarsTemplate == "" {
                                        // No templating so we stop here (further matches would just
                                        // produce the same labels / vars)
                                        break
                                }
```

Reply:
sure, will do